src/frontend/src/utils/condition.rs

```diff
@@ -74,49 +74,29 @@ impl Condition {
     }
 
     pub fn to_expr(self) -> ExprImpl {
-        let mut iter = self.conjunctions.into_iter();
-        if let Some(mut ret) = iter.next() {
-            for expr in iter {
-                ret = FunctionCall::new(ExprType::And, vec![ret, expr])
-                    .unwrap()
-                    .into();
-            }
-            ret
-        } else {
-            Literal::new(Some(ScalarImpl::Bool(true)), DataType::Boolean).into()
-        }
+        merge_expr_by_binary(
+            self.conjunctions.into_iter().map(Cow::Owned),
+            ExprType::And,
+            ExprImpl::literal_bool(true),
+        )
     }
 
     // TODO(TaoWu): We might also use `Vec<ExprImpl>` form of predicates in compute node,
     // rather than using `AND` to combine them.
     pub fn as_expr(&self) -> ExprImpl {
-        let mut iter = self.conjunctions.iter();
-        if let Some(e) = iter.next() {
-            let mut ret = e.clone();
-            for expr in iter {
-                ret = FunctionCall::new(ExprType::And, vec![ret, expr.clone()])
-                    .unwrap()
-                    .into();
-            }
-            ret
-        } else {
-            Literal::new(Some(ScalarImpl::Bool(true)), DataType::Boolean).into()
-        }
+        merge_expr_by_binary(
+            self.conjunctions.iter().map(Cow::Borrowed),
```

Comment:
Why not directly `cloned()` here? Seems we are always calling `into_owned` on `Cow`s.